tsconfig.json

```
@@ -2,24 +2,31 @@
 	"compilerOptions": {
 		"allowJs": true,
 		"checkJs": true,
-		"noEmit": true,
-		"module": "es2015",
+		"emitDeclarationOnly": true,
+		"declaration": true,
+		"module": "esnext",
 		"moduleResolution": "node",
-		"target": "es5",
-		"baseUrl": "packages",
-		"paths": {
-			"@tannin/*": [ "*/index.js" ],
-			"tannin": [ "tannin/index.js" ]
-		}
+		"target": "esnext",
+		"composite": true,
+
+		"strict": true,
+		"strictNullChecks": true,
+		"noImplicitAny": true,
+		"noUnusedLocals": true,
+		"noUnusedParameters": true,
+		"noImplicitReturns": true,
+		"noFallthroughCasesInSwitch": true,
+
+		/* This needs to be false so our types are possible to consume without setting this */
+		"esModuleInterop": false,
+		"isolatedModules": true
 	},
-	"include": [
-		"packages"
-	],
+	"include": [],
```

Trying to consider how this configuration would be extended, I wondering if we ought to be consistent between this and extending configurations toward one or the other of `files` or `include`, as a reflection of how those sub-configurations are extending this. Or to just omit it altogether here.

---

The other projects are using `files`, so it makes sense to use that here 👍 

My thinking here is that the root is really an empty project, it shouldn't include anything.

There's an interesting demo project that can serve as inspiration:

https://github.com/RyanCavanaugh/project-references-demo

```suggestion
	"files": [],
```